internal/docstore/doc.go

```diff
@@ -13,38 +13,142 @@
+// memdocstore, an in-process, in-memory implementation suitable for testing and
+// development.
 //
+// Your application should import one of these provider-specific subpackages and use
+// its exported functions to create a *Collection; do not use the NewCollection
+// function in this package. For example:
 //
-// Representing Times
+//   coll, err := memdocstore.OpenBucket("SSN", nil)
+//     if err != nil {
+//       return fmt.Errorf("opening collection: %v", err)
+//     }
+//     // coll is a *docstore.Collection
 //
-// Docstore can store and retrieve values of type time.Time, with two caveats. First,
-// the timezone may not be preserved. Second, Docstore guarantees only that time.Time
-// values are represented to millisecond precision. Many providers will do better,
-// but if you need to be sure that times are stored with nanosecond precision,
-// convert the time.Time to another type before storing, and re-create when you
-// retrieve it. For instance, you could call the time's UnixNano method to get an
-// int64, and get the original time back (in the local timezone) with the time.Unix
-// function.
+// Then, write your application code using the *Collection type. You can easily
+// reconfigure your initialization code to choose a different provider.
+// You can develop your application locally using memdocstore, or deploy it to
+// multiple Cloud providers. You may find http://github.com/google/wire useful
+// for managing your initialization code.
+//
+// Alternatively, you can construct a *Collection via a URL and the OpenCollection
+// function in this package. See https://gocloud.dev/concepts/urls for more
+// information.
+//
+//
+// Documents
+//
+// Docstore allows you to represent documents as either map[string]interface{} or
+// struct pointers. Using structs is recommended, because it enforces some structure
+// on your data. Docstore mimics the encoding/json package in its treatment of
+// structs: by default, a struct's exported fields are the fields of the document.
+// You can alter this default mapping by using a struct tag beginning with
+// "docstore:". Docstore struct tags support renaming, omitting fields
+// unconditionally, or omitting them only when they are empty, exactly like
+// encoding/json.
+//
+//
+// Representing Data
+//
+// The value stored in a document field can be any of a wide range of types. All
+// primitive types except for complex numbers are supported, as well as slices and
+// maps (the map key type must be a string, an integer, or a type that implements
+// encoding.TextMarshaler). In addition, any type that implements
+// encoding.BinaryMarshaler or encoding.TextMarshaler is permitted. This set of types
+// closely matches the encoding/json package.
+//
+// Times deserve special mention. Docstore can store and retrieve values of type
+// time.Time, with two caveats. First, the timezone may not be preserved. Second,
```

Comment:
Maybe we should make this "will not be preserved"?

Reply:
done